From ee8de1a11faa409987b409d92410d3b1c66b3304 Mon Sep 17 00:00:00 2001
From: Miklos Csanady <miklos.csanady@cloudera.com>
Date: Fri, 21 Apr 2017 10:31:13 +0200
Subject: [PATCH 1075/1164] Revert "CDH-43527: HIVE-16047: Shouldn't try to
 get KeyProvider unless encryption is enabled (Rui
 reviewed by Xuefu and Ferdinand)"

This reverts commit 7f8e9e83794424e70483ef741503ec4b16feef71.

Change-Id: Ie6022ba1a494f826b4a864e536be0d8582ac2073
---
 .../apache/hadoop/hive/shims/Hadoop23Shims.java    |   16 +---------------
 1 file changed, 1 insertion(+), 15 deletions(-)

diff --git a/shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java b/shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java
index 9d1fd32..360304b 100644
--- a/shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java
+++ b/shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java
@@ -61,8 +61,6 @@
 import org.apache.hadoop.fs.permission.AclStatus;
 import org.apache.hadoop.fs.permission.FsAction;
 import org.apache.hadoop.fs.permission.FsPermission;
-import org.apache.hadoop.hdfs.DFSClient;
-import org.apache.hadoop.hdfs.DFSConfigKeys;
 import org.apache.hadoop.hdfs.DistributedFileSystem;
 import org.apache.hadoop.hdfs.MiniDFSCluster;
 import org.apache.hadoop.hdfs.protocol.HdfsConstants;
@@ -1261,22 +1259,10 @@ public HdfsEncryptionShim(URI uri, Configuration conf) throws IOException {
       DistributedFileSystem dfs = (DistributedFileSystem)FileSystem.get(uri, conf);
 
       this.conf = conf;
-      this.keyProvider = isEncryptionEnabled(dfs.getClient(), dfs.getConf()) ?
-          dfs.getClient().getKeyProvider() : null;
+      this.keyProvider = dfs.getClient().getKeyProvider();
       this.hdfsAdmin = new HdfsAdmin(uri, conf);
     }
 
-    private boolean isEncryptionEnabled(DFSClient client, Configuration conf) {
-      try {
-        DFSClient.class.getMethod("isHDFSEncryptionEnabled");
-      } catch (NoSuchMethodException e) {
-        // the method is available since Hadoop-2.7.1
-        // if we run with an older Hadoop, check this ourselves
-        return !conf.getTrimmed(DFSConfigKeys.DFS_ENCRYPTION_KEY_PROVIDER_URI, "").isEmpty();
-      }
-      return client.isHDFSEncryptionEnabled();
-    }
-
     @Override
     public boolean isPathEncrypted(Path path) throws IOException {
       Path fullPath;
-- 
1.7.9.5

